volumes:
  dbbackups:
  postgis-data:
  raster-db:
  auth:


services:
  web-server:
    build: ./containers/web-server
    volumes:
      - auth:/etc/nginx/auth
    ports:
      - "80:80"
      - "443:443"
    restart: on-failure

  db:
    image: kartoza/postgis:14-3.1
    volumes:
      - postgis-data:/var/lib/postgresql
    environment:
      - POSTGRES_DB=jamaica
      - POSTGRES_USER=docker
      - POSTGRES_PASS=docker
      - ALLOW_IP_RANGE=0.0.0.0/0
      - POSTGRES_MULTIPLE_EXTENSIONS=postgis
    expose:
      - 5432
    restart: on-failure
    healthcheck:
      test: "exit 0"

  backend:
    build: ./containers/backend
    environment:
      - PGHOST=db
      - PGDATABASE=jamaica
      - PGUSER=docker
      - PGPASSWORD=docker
    expose:
      - 8888
    restart: on-failure

  raster-tileserver:
    build: ./containers/raster
    volumes:
      - raster-db:/opt/terracotta/db
      - ./mounts/raster:/opt/terracotta/data # bind mount
    expose:
      - 5000
    restart: on-failure

  vector-tileserver:
    build: ./containers/vector
    volumes:
      - ./mounts/vector:/data # bind mount
    expose:
      - 8080
    restart: on-failure
